lib/mllp/client.ex

```diff
@@ -104,13 +167,40 @@ defmodule MLLP.Client do
 
   def format_error(err), do: inspect(err)
 
+  @doc """
+  Starts a new MLLP.Client.
+
+  MLLP.Client.start_link/4 will start a new MLLP.Client process.
+
+  This function will raise a `ArgumentError` if an invalid `ip_address()` is provided. 
+
+  ## Options
+
+  * `:auto_reconnect_interval` - Specify the interval between connection attempts. Specifically, if an attempt
+     to establish a connection fails, either post-init or at some point during the life span of the client, the value
+     of this option shall determine how often to retry a reconnection. Defaults to 1000 milliseconds.
+
+  * `:reply_timeout` - Optionally specify a timeout value for receiving a response. Must be a positive integer or 
+     `:infinity`. Defaults to `:infinity`.
+    
+  * `:socket_opts` -  A list of socket options as supported by [`:gen_tcp`](`:gen_tcp`). 
```

Comment:
double `[:gen_tcp]`

Reply:
Ahh, this is actually just link i.e., `:gen_tcp` ends up pointing to the `:gen_tcp` module document.